source/shadow.c

```diff
@@ -320,14 +456,22 @@ static ShadowStatus_t extractShadowRootAndName( const char * pTopic,
                                      topicLength - *pConsumedTopicLength,
                                      pShadowNameLength );
 
-        if( shadowStatus == SHADOW_SUCCESS )
+        if( shadowStatus != SHADOW_SUCCESS )
         {
```

Comment:
See my previous comment about the initialization of `shadowStatus` in these functions; we should also do this at line 441.